src/main/java/picard/sam/BamErrorMetric/ReadBaseStratification.java

```diff
@@ -0,0 +1,909 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2018 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package picard.sam.BamErrorMetric;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMTag;
+import htsjdk.samtools.util.Lazy;
+import htsjdk.samtools.util.SamLocusIterator;
+import htsjdk.samtools.util.SequenceUtil;
+import org.broadinstitute.barclay.argparser.CommandLineParser;
+import picard.sam.markduplicates.util.OpticalDuplicateFinder;
+import picard.sam.util.PhysicalLocation;
+import picard.sam.util.PhysicalLocationInt;
+
+import java.util.concurrent.ExecutionException;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * Classes, methods, and enums that deal with the stratification of read bases and reference information.
+ *
+ * @author Yossi Farjoun
+ */
+public class ReadBaseStratification {
+    // This variable has to be set using setLongHomopolymer _before_ the first call to binnedHomopolymerStratifier.get()
+    // if you need different binned Homopolymers with different LONG_HOMOPOLYMER values, you'll need to make your own
+    // instances.
+    private static int LONG_HOMOPOLYMER = 6;
+    private static int GC_CACHE_SIZE = 1000;
+
+
+    /* ***** SETTERS ********** */
+
+    public static void setGcCacheSize(int gcCacheSize) {
+        GC_CACHE_SIZE = gcCacheSize;
+    }
+
+    /**
+     * defaults to 6
```

Comment:
`defaults to 6` does not belong here, but perhaps at the class level.  The doc should tell us what this method does.